### ISSUE TITLE
formula for advantages in GRPO

### ISSUE THREAD

**Issue by catalin-hanga** [thinking-machines-lab/tinker-cookbook#579]
**Page:** https://tinker-docs.thinkingmachines.ai/tutorials/advanced/rl-hyperparams/

**Issue:**

In the online documentation and in the cookbook code example, the advantages are centered within each group:
advantage_i = reward_i - mean(rewards in group)

However, in the original GRPO paper from DeepSeek, the advantage was normalized:
advantage_i = (reward_i - mean(rewards in group)) / std(rewards in group)

Is there a reason why in the documentation the advantages are not divided by the standard deviation of the rewards within each group ?

**Comment by YujiaBao** [comment]
Thanks for the careful read! This is intentional — we use mean-centering without std normalization:

```
advantage_i = reward_i - mean(rewards in group)
```

The original DeepSeek GRPO paper does normalize by std, but subsequent work has shown this introduces bias. In particular, [*Understanding R1-Zero-Like Training: A Critical Perspective*](https://arxiv.org/abs/2503.20783) (Liu et al., 2025) identifies two problems with dividing by std:

1. **Difficulty bias** — questions where rewards have low variance (too easy or too hard) get disproportionately large gradients because `1/std` blows up when rewards within a group are similar.
2. **Length bias** — combined with per-token normalization, std normalization causes incorrect responses to grow longer during training.

Their proposed fix (Dr. GRPO) is exactly what we implement: `reward - mean(rewards)`, which recovers an unbiased policy gradient objective.

We'll update the tutorial text to call out this difference from the original paper more explicitly. Thanks for raising it!
